### PR TITLE
types(schema): support `required: { isRequired: true }` syntax in schema definition

### PR DESCRIPTION
**Pull request by vkarpov15**
Re: #13514

**Summary**

One potential workaround for #13514 is using object syntax for defining `required`, because the following works fine:

```javascript
new Schema({
  email: {
    type: String,
    required: { isRequired: true, message: 'Email is required' }
  }
});
```

But we don't currently support that in TypeScript. This PR adds support for the `required: { isRequired: true }` syntax. Unfortunately doesn't help very much with #13514.

**Examples**

### CODE REVIEW

File: test/types/schema.test.ts [hunk 1]
```diff
@@ -1134,3 +1134,16 @@ function maps() {
   expectType<Map<string, number>>(doc.myMap);
   expectType<number | undefined>(doc.myMap!.get('answer'));
 }
+
+function gh13514() {
+  const schema = new Schema({
+    email: {
+      type: String,
+      required: [true, 'email is required']
```
Comment (hasezoey) [inline test/types/schema.test.ts:1142]: if i understand the addition correctly, this test does not actually cover that

**Review by hasezoey — COMMENTED** [review]
Change looks good to me, but the test not really

File: test/types/schema.test.ts (hunk 1 above)
Comment (vkarpov15) [reply test/types/schema.test.ts:1142]: Yeah this test is attempting to repro #13514 using tsd, no luck.

**Review by IslandRhythms — APPROVED** [review]
LGTM

**Pull request merged**